iis/tests/test_iis.py

```diff
@@ -81,30 +95,40 @@ def test_check(aggregator):
     c.check(None)
     iis_host = c.get_iishost()
 
-    # Test metrics
-    # ... normalize site-names
-    default_site_name = re.sub(r"[,\+\*\-/()\[\]{}\s]", "_", WIN_SERVICES_CONFIG['sites'][0])
-    ok_site_name = re.sub(r"[,\+\*\-/()\[\]{}\s]", "_", WIN_SERVICES_CONFIG['sites'][1])
-    fail_site_name = re.sub(r"[,\+\*\-/()\[\]{}\s]", "_", WIN_SERVICES_CONFIG['sites'][2])
-
-    for site_name in [default_site_name, ok_site_name, 'Total']:
-        for metric_def in DEFAULT_COUNTERS:
-            mname = metric_def[3]
-            aggregator.assert_metric(mname, tags=["mytag1", "mytag2", "site:{0}".format(site_name), iis_host], count=1)
-
-        aggregator.assert_service_check(
-            'iis.site_up', status=IIS.OK, tags=["mytag1", "mytag2", "site:{0}".format(site_name), iis_host], count=1
-        )
-
-    aggregator.assert_service_check(
-        'iis.site_up',
-        status=IIS.CRITICAL,
-        tags=["mytag1", "mytag2", "site:{0}".format(fail_site_name), iis_host],
-        count=1,
-    )
-
-    # Check completed with no warnings
-    # self.assertFalse(logger.warning.called)
+    # Test tag name normalization
+    sites = ['Total']
+    for site in instance['sites']:
+        sites.append(re.sub(r'[,+*\-/()\[\]{}\s]', '_', site))
+    app_pools = ['Total']
+    for app_pool in instance['app_pools']:
+        app_pools.append(re.sub(r'[,+*\-/()\[\]{}\s]', '_', app_pool))
+
+    namespace_data_ok = ((SITE_METRICS, IIS.SITE, sites[:-1]), (APP_POOL_METRICS, IIS.APP_POOL, app_pools[:-1]))
+    for metrics, namespace, values in namespace_data_ok:
+        for metric in metrics:
+            for value in values:
+                aggregator.assert_metric(
+                    metric, tags=['mytag1', 'mytag2', '{}:{}'.format(namespace, value), iis_host], count=1
+                )
+
+    for _, namespace, values in namespace_data_ok:
+        for value in values[1:]:
+            aggregator.assert_service_check(
+                'iis.{}_up'.format(namespace),
+                IIS.OK,
+                tags=['mytag1', 'mytag2', '{}:{}'.format(namespace, value), iis_host],
+                count=1,
+            )
+
+    namespace_data_failed = ((SITE_METRICS, IIS.SITE, sites[-1:]), (APP_POOL_METRICS, IIS.APP_POOL, app_pools[-1:]))
```

Comment:
The use of slices (`sites[:-1]`, `values[1:]`, `sites[-1:]`) in this test make it harder to understand. I feel there is maybe a better way.